<?xml version="1.0" encoding="UTF-8" ?>
<testsuite tests="70" failures="0" errors="9" skipped="0" time="3.351" name="file &quot;oauth2.app&quot;">
  <testcase time="0.001" name="oauth2_tests:52 bad_authorize_password_test_">
    <error type="error">
::in function oauth2_mock_backend:jwt_issuer/0
  called as jwt_issuer()
in call from oauth2:auth_user/3 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 450)
in call from oauth2_tests:'-bad_authorize_password_test_/0-fun-1-'/0 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 53)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 71)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 522)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 347)
in call from eunit_proc:handle_test/2 (eunit_proc.erl, line 505)
in call from eunit_proc:tests_inorder/3 (eunit_proc.erl, line 447)
**error:undef
    </error>
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:57 bad_authorize_password_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:62 bad_authorize_password_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:67 bad_authorize_password_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:72 bad_authorize_password_test_">
    <error type="error">
::in function oauth2_mock_backend:jwt_issuer/0
  called as jwt_issuer()
in call from oauth2:auth_user/3 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 450)
in call from oauth2:authorize_password/4 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 114)
in call from oauth2_tests:'-bad_authorize_password_test_/0-fun-9-'/0 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 73)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 71)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 522)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 347)
in call from eunit_proc:handle_test/2 (eunit_proc.erl, line 505)
**error:undef
    </error>
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:78 bad_authorize_password_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:84 bad_authorize_password_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:90 bad_authorize_password_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:96 bad_authorize_password_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:0 authorize_implicit_grant_test_">
    <error type="error">
::in function oauth2_mock_backend:jwt_issuer/0
  called as jwt_issuer()
in call from oauth2:auth_user/3 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 450)
in call from oauth2:authorize_password/5 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 134)
in call from oauth2_tests:'-authorize_implicit_grant_test_/0-fun-1-'/0 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 113)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 71)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 522)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 347)
in call from eunit_proc:handle_test/2 (eunit_proc.erl, line 505)
**error:undef
    </error>
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:134 bad_authorize_client_credentials_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:140 bad_authorize_client_credentials_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:145 bad_authorize_client_credentials_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:151 bad_authorize_client_credentials_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.123" name="oauth2_tests:0 bad_ttl_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:0 bad_ttl_test_">
    <error type="error">
::in function oauth2_mock_backend:jwt_issuer/0
  called as jwt_issuer()
in call from oauth2:auth_user/3 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 450)
in call from oauth2:authorize_code_request/5 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 197)
in call from oauth2_tests:issue_access_code/1 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 426)
in call from oauth2_tests:'-bad_ttl_test_/0-fun-3-'/0 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 178)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 71)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 522)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 347)
**error:undef
    </error>
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:0 bad_ttl_test_">
    <error type="error">
::in function oauth2_mock_backend:jwt_issuer/0
  called as jwt_issuer()
in call from oauth2:auth_user/3 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 450)
in call from oauth2:authorize_code_request/5 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 197)
in call from oauth2_tests:issue_access_code/1 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 426)
in call from oauth2_tests:'-bad_ttl_test_/0-fun-5-'/0 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 187)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 71)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 522)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 347)
**error:undef
    </error>
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:0 verify_access_token_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:217 verify_access_token_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:0 bad_access_code_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:0 verify_access_code_test_">
    <error type="error">
::in function oauth2_mock_backend:jwt_issuer/0
  called as jwt_issuer()
in call from oauth2:auth_user/3 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 450)
in call from oauth2:authorize_code_request/5 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 197)
in call from oauth2_tests:issue_access_code/1 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 426)
in call from oauth2_tests:'-verify_access_code_test_/0-fun-3-'/0 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 269)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 71)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 522)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 347)
**error:undef
    </error>
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:0 bad_refresh_token_test_">
    <error type="error">
::in function oauth2_mock_backend:jwt_issuer/0
  called as jwt_issuer()
in call from oauth2:auth_user/3 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 450)
in call from oauth2:authorize_code_request/5 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 197)
in call from oauth2_tests:'-bad_refresh_token_test_/0-fun-4-'/0 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 299)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 71)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 522)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 347)
in call from eunit_proc:handle_test/2 (eunit_proc.erl, line 505)
**error:undef
    </error>
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:0 verify_refresh_token_test_">
    <error type="error">
::in function oauth2_mock_backend:jwt_issuer/0
  called as jwt_issuer()
in call from oauth2:auth_user/3 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 450)
in call from oauth2:authorize_code_request/5 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 197)
in call from oauth2_tests:issue_access_code/1 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 426)
in call from oauth2_tests:'-verify_refresh_token_test_/0-fun-1-'/0 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 352)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 71)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 522)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 347)
**error:undef
    </error>
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_tests:0 verify_refresh_token_test_">
    <error type="error">
::in function oauth2_mock_backend:jwt_issuer/0
  called as jwt_issuer()
in call from oauth2:auth_user/3 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 450)
in call from oauth2:authorize_password/4 (/Users/moritz/Projects/kivra/oauth2/src/oauth2.erl, line 114)
in call from oauth2_tests:issue_token_and_refresh_with_user_name_and_password/2 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 448)
in call from oauth2_tests:'-verify_refresh_token_test_/0-fun-4-'/1 (/Users/moritz/Projects/kivra/oauth2/test/oauth2_tests.erl, line 368)
in call from lists:foreach/2 (lists.erl, line 1342)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 71)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 522)
**error:undef
    </error>
    <system-out>
    </system-out>
  </testcase>
  <testcase time="1.507" name="oauth2_priv_set_tests:37 proper_type_spec_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:43 new_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:46 new_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:49 new_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:52 new_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:59 is_subset_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:69 is_subset_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:72 is_subset_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:75 is_subset_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:78 is_subset_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:81 is_subset_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:84 is_subset_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_priv_set_tests:87 is_subset_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.757" name="oauth2_response_tests:46 proper_type_spec_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:56 new_1_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:57 new_1_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:58 new_1_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:59 new_1_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:69 new_2_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:70 new_2_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:71 new_2_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:72 new_2_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:82 new_4_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:83 new_4_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:84 new_4_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:85 new_4_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:95 new_5_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:96 new_5_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:97 new_5_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:98 new_5_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:109 new_6_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:110 new_6_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:111 new_6_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:112 new_6_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:113 new_6_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:0 access_token_test">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:0 access_code_test">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:0 expires_in_test">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:0 scope_test">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:0 refresh_token_test">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:0 resource_owner_test">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_response_tests:0 token_type_test">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.021" name="oauth2_response_tests:0 to_proplist_test">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.021" name="oauth2_response_tests:0 to_map_test">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.685" name="oauth2_token_tests:37 proper_type_spec_test_">
    <system-out>
    </system-out>
  </testcase>
  <testcase time="0.000" name="oauth2_token_tests:0 generate_test">
    <system-out>
    </system-out>
  </testcase>
</testsuite>
